modules/local/contig_filter/main.nf

```diff
@@ -1,38 +1,37 @@
 process CONTIG_FILTER {
-    conda "${moduleDir}/environment.yml"
-    container "wave.seqera.io/wt/67f4c80bd6c6/wave/build:contig_filter--f8e5cec565adc43e"
+    // conda "${moduleDir}/environment.yml"
+    // container "wave.seqera.io/wt/67f4c80bd6c6/wave/build:contig_filter--f8e5cec565adc43e"
 
     input:
-    tuple val(meta), path("*.tsv")
+    tuple val(meta), path(tsv)
     
     output:
     tuple val(meta), path("*.tsv")  , emit: viral_contigs_metadata
     path "versions.yml"             , emit: versions
     
     script:
-    //how does lineage file here link to the input file
+    //how does lineage file here link to the input file 
     //should these files be allowed the option of empty
-    def lineage_file = task.ext.lineage_file ?: ''
-    def viral_contigs_metadata = task.ext.viral_contigs_metadata ?: ''
+    def viral_contigs_metadata = task.ext.viral_contigs_metadata 
     """
     #contig filtering according to kingdom viruses
     #extract contig matches that are part of viruses
     while IFS=\$'\\t' read -r col1 col2 col3 col4 rest; do
         if [[ \${col4} == *Virus* ]]; then
             echo -e "\${col1}\\t${col2}\\t${col4}\\t${rest}" >> "${viral_contigs_metadata}"
         fi
-    done < ${lineage_file}
+    done < ${tsv}
     """
 
     stub:
-    def lineage_file = task.ext.lineage_file ?: ''
-    def viral_contigs_metadata = task.ext.viral_contigs_metadata ?: ''
+
+    def viral_contigs_metadata = task.ext.viral_contigs_metadata 
     """
-    touch ${viral_contigs_metadata}
+    touch viral_contigs_metadata.tsv
     
     cat <<-END_VERSIONS > versions.yml
     "${task.process}":
-        contig_filter: \$(contig_filter -v 2>&1 | sed 's/CONTIG_FILTER v//')
+        contig_filter: "1.0.0"
```

Comment:
Ideally, hard-coding version isn't the best practice however in the interest of time we can skim this 👍